### DISCUSSION TITLE
Open other apps in app window

### DISCUSSION THREAD

**Discussion by gyanpatel** [fyne-io/fyne#3583] [General]
Is it possible to open other apps in a window ? e.g. I want to create an app which can open multiple apps within itself.
Like separate tab windows for various social media apps. Please advise.
Many thanks

**Comment by Jacalz** [comment]
Your Fyne app that you create at the application start has a method called `NewWindow(title string)` (the very first you use when creating the initial/first window).

This can be used more times to create separate windows. The only difference in that case is that you want to call `.Show()` instead of `ShowAndRun()` for later windows. The reasons for it is that the application already is running when you show more than one window.

**Reply by gyanpatel** [reply]
The question is whether the window can allow to open other mobile apps within it ? 

Thank you for your response.

**Reply by Bluebugs** [reply]
If I understand what you want to do, you want to have a widget which has its content being another application window. This is what a compositor does and we do not have currently that capability in Fyne. It is something that would be nice to have.

**Reply by andydotxyz** [reply]
The specific mention of mobile however makes it sound impossible. The OS sandboxing won't allow it even if we had the technology in place.

**Reply by Bluebugs** [reply]
In the space of Android and Linux, it might be possible with a very large amount of work. Project like SPURV ( https://memcpy.io/running-android-next-to-wayland.html ) have shown that it is possible to get an Android application to run on a normal linux and get its content display on a Wayland compositor.  SPURV never got any traction, so it would require to revive it today. Then it would be necessary to add Wayland compositor capability to Fyne and you would finally have something. That will involve a very large amount of work.

**Reply by andydotxyz** [reply]
I think it would be reasonable to assume that the OP is expecting to work on a vanilla mobile OS/runtime ;)
